easybuild/easyblocks/m/mcr.py

```diff
@@ -0,0 +1,146 @@
+                regdest = re.compile(r"^# destinationFolder=.*", re.M)
+                regagree = re.compile(r"^# agreeToLicense=.*", re.M)
+                regmode = re.compile(r"^# mode=.*", re.M)
+
+                config = regdest.sub("destinationFolder=%s" % self.installdir, config)
+                config = regagree.sub("agreeToLicense=Yes", config)
+                config = regmode.sub("mode=silent", config)
+            else:
+                config = "destinationFolder=%s\n" % self.installdir
+                config += "agreeToLicense=Yes\n"
+                config += "mode=silent\n"
+
+            f = open(configfile, 'w')
+            f.write(config)
+            f.close()
+
+        except IOError, err:
+            raise EasyBuildError("Failed to create installation config file %s: %s", configfile, err)
+
+        self.log.debug('configuration file written to %s:\n %s' % (configfile, config))
+
+    def build_step(self):
+        """No building of MCR, no sources available."""
+        pass
+
+    def install_step(self):
+        """MCR install procedure using 'install' command."""
+
+        src = os.path.join(self.cfg['start_dir'], 'install')
+
+        # make sure install script is executable
+        try:
+            if os.path.isfile(src):
+                self.log.info("Doing chmod on source file %s" % src)
+                os.chmod(src, 0755)
+            else:
+                self.log.info("Did not find source file %s" % src)
+        except OSError, err:
+            raise EasyBuildError("Failed to chmod install script: %s", err)
+
+        # make sure $DISPLAY is not defined, which may lead to (hard to trace) problems
+        # this is a workaround for not being able to specify --nodisplay to the install scripts
+        if 'DISPLAY' in os.environ:
+            os.environ.pop('DISPLAY')
+
+        if not '_JAVA_OPTIONS' in self.cfg['preinstallopts']:
+            self.cfg['preinstallopts'] = ('export _JAVA_OPTIONS="%s" && ' % self.cfg['java_options']) + self.cfg['preinstallopts']
+        configfile = "%s/%s" % (self.builddir, self.configfilename)
+        cmd = "%s ./install -v -inputFile %s %s" % (self.cfg['preinstallopts'], configfile, self.cfg['installopts'])
+        run_cmd(cmd, log_all=True, simple=True)
+
+    def sanity_check_step(self):
+        """Custom sanity check for MCR."""
+
+        custom_paths = {
+                        'files': [''],
+                        'dirs': ['%s/%s/glnxa64' % (self.extradir[self.cfg['version']], x) for x in ['runtime' , 'bin', 'sys/os']],
+                       }
+
+        super(EB_MCR, self).sanity_check_step(custom_paths=custom_paths)
+
+    def make_module_extra(self):
+        """Extend PATH and set proper _JAVA_OPTIONS (e.g., -Xmx)."""
+
+        txt = super(EB_MCR, self).make_module_extra()
+
+        txt += self.module_generator.set_environment('XAPPLRESDIR', os.path.join(self.installdir, self.extradir[self.cfg['version']], 'X11', 'app-defaults'))
```

Comment:
It's OK to make the mapping from version to subdir in the easyblock, but to make this easyblock compatible with `--module-only` and avoid hard crashes if a new version is attempted, use `self.extradir.get` rather than `self.extradir[..]`